### DISCUSSION TITLE
How to override one specific color of a scheme ?

### DISCUSSION THREAD

**Discussion by bro69noscope** [wezterm/wezterm#6061] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I am aware that ...

```lua
local wezterm = require 'wezterm'

local scheme = wezterm.color.get_builtin_schemes()['Gruvbox Light']
scheme.background = 'red'

return {
  color_schemes = {
    -- Override the builtin Gruvbox Light scheme with our modification.
    ['Gruvbox Light'] = scheme,

    -- We can also give it a different name if we don't want to override
    -- the default
    ['Gruvbox Red'] = scheme,
  },
  color_scheme = 'Gruvbox Light',
}
```

... Is a thing but it does not seem to cover my use case at all. How do I change a value in the scheme that is associated with a specific syntax higlight ? Like change the color of whatever highlight group would be assigned for pwsh args like -Recurse or -Force ?

**Comment by bew** [comment]
I think you are confused, a terminal like wezterm has no knowledge of what is being displayed and does not have syntax highlight groups, it just knows about cells and their current color based on what the running program told the terminal how these cells should be printed at the time.

So in this case it is Powershell that defines that some part of its output should be colored in X color, and when printing a folder name for example powershell just tells the terminal "for the next stuff I print, color it in that specific blue", then after the name powershell tells the terminal "for the next stuff, reset the color".

I'd suggest you look into configuration of Powershell if you want to change how the output is colored.
(although I don't think powershell has much configuration for that kind of things..)

_(hopefully it helps you see the difference, don't hesitate to ask further question)_

**Comment by bro69noscope** [comment]
Oh okay I was applying a flawed logic to this. Thanks a lot for pointing it out ! So these are the value I want to change right:

```ps1
OperatorColor                          : "`e[90m"
ParameterColor                         : "`e[90m"
```

Which afaik are the ones who correspond to "bright black" ? Will look into replacing these values only for my current theme and report.

**Reply by bro69noscope** [reply, marked as answer]
Ok so figured it out. For someone else who would want to do this...
1. `get-PSReadLineOption` in pwsh will show you the ansi espace codes for the associated powershell syntax. (Same output as above)
2. The table below will show matches between ansi sequences and "WezTerm colors"
![image](https://github.com/user-attachments/assets/54fe5e95-014b-4cd2-86c4-a16debe04f84)

We can then set cherry picked colors this way:
```lua
local scheme_name = "tokyonight_moon"
local scheme = wezterm.color.get_builtin_schemes()[scheme_name]

config.color_scheme = scheme_name
scheme.brights[1] = "#ff69b4" -- Set bright black to bright pink for testing (or if u like pink, its ok to like pink)
config.colors = scheme
```
And if you want custom tab colors, like I do too, I think we then have to use:
```lua
config.colors.tab_bar = {
	background = "#16161e"
	active_tab = {
	    -- code
	},
	-- etc ..
}
```

because using:
```lua
config.colors = {
    tab_bar = {
     -- code
    }
}
```
... Like I initially did seems to not take the previous color setting in account. Altough I'm not sure and don't really understand what invoking the same key does to lua table (merge or override). Probably could still do it with some extend syntax or wathever, idk lua, I'm learning.

If this is useful to someone then cool. :heart:

**Comment by Alien-lv426** [comment]
Out of interest, why not use

 `Set-PSReadLineOption -Colors @{}` 

without needing to involve wezterm?

**Reply by bro69noscope** [reply]
Unaware of that option lol 

Without looking into it I would have  maybe assumed intuitively that colors set by pwsh this way would be overridden by the wezterm scheme. Which is prob not the case if they are set to RGB hex ?

**Reply by bew** [reply]
Yeah if you set rgb colors in powershell config' it will just use that color and you don't need to tweak colors in terminal's config

**Reply by Alien-lv426** [reply]
For example:

`Set-PSReadLineOption -Colors @{
    Operator = 'Red'
    Parameter = 'Yellow'
}`
